SetupDataPkg/Library/ConfigVariableListLib/ConfigVariableListLib.c

```diff
@@ -67,27 +280,18 @@ ParseActiveConfigVarList (
 
   *ConfigVarListCount = 0;
 
-  // Populate the slot with default one from FV.
-  Status = GetSectionFromAnyFv (
-             (EFI_GUID *)PcdGetPtr (PcdSetupConfigActiveProfileFile),
-             EFI_SECTION_RAW,
-             0,
-             &BlobPtr,
-             &BinSize
-             );
-
-  if (EFI_ERROR (Status) || (BlobPtr == NULL) || (BinSize == 0)) {
-    DEBUG ((DEBUG_ERROR, "%a Failed to read active profile data (%r)\n", __FUNCTION__, Status));
-    ASSERT (FALSE);
-    Status = EFI_NOT_FOUND;
+  if ((VariableListBuffer == NULL) || (VariableListBufferSize == 0)) {
+    DEBUG ((DEBUG_ERROR, "%a Incoming variable list buffer (base: %p, size: 0x%x) invalid\n", __FUNCTION__, VariableListBuffer, VariableListBufferSize));
+    *ConfigVarListPtr = NULL;
+    Status            = EFI_INVALID_PARAMETER;
     goto Exit;
   }
 
   if (ConfigVarName == NULL) {
     // We don't know how many entries there are, for now allocate far too large of a pool, we'll realloc once we
     // know how many entries there are. We are returning full list of entries.

```

Comment:
comment is out of date now, we aren't allocating too large of a pool anymore

Reply:
Updated comment.